packages/toolkit/src/codewhisperer/service/transformByQHandler.ts

```diff
@@ -375,26 +375,19 @@ function getFilesRecursively(dir: string, isDependenciesFolder: boolean): string
     return files
 }
 
-/* TO-DO: consider combining installProjectDepedencies() and copyProjectDependencies() into 1 function runMavenCommand()
- * do this once we fully confirm that we definitely want to run "mvn install"
- * also, consider either notifying user how to set JAVA_HOME if mvn install fails, or doing it for them?
- */
-function installProjectDependencies(buildCommand: CodeTransformMavenBuildCommand, modulePath: string) {
-    let baseCommand = buildCommand as string
-    if (baseCommand === 'mvnw') {
-        baseCommand = './mvnw'
-        if (os.platform() === 'win32') {
-            baseCommand = './mvnw.cmd'
-        }
-        const executableName = baseCommand.slice(2) // remove the './' part
-        const executablePath = path.join(modulePath, executableName)
-        if (!fs.existsSync(executablePath)) {
-            throw new ToolkitError('Maven Wrapper not found', { code: 'MavenWrapperNotFound' })
-        }
+// run 'install' with either 'mvnw.cmd', './mvnw', or 'mvn' (if wrapper exists, we use that, otherwise we use regular 'mvn')
+function installProjectDependencies() {
+    // baseCommand will be one of: 'mvnw.cmd', './mvnw', 'mvn'
+    let baseCommand = transformByQState.getMavenName()
+    // if JAVA_HOME could not be found or did not match the project JDK, baseCommand is prefixed with JAVA_HOME={user-input}
+    if (transformByQState.getJavaHome() !== undefined) {
+        baseCommand = `JAVA_HOME=${transformByQState.getJavaHome()} ${baseCommand}`
     }
+    const modulePath = transformByQState.getProjectPath()
 
     transformByQState.appendToErrorLog(`Running command ${baseCommand} clean install`)
 
+    // Note: IntelliJ runs 'clean' separately from 'install'. Evaluate benefits (if any) of this.
```

Comment:
I personally don't think there are any benefits to doing this. Can change later if needed but all `clean` does is delete the "target" directory so doesn't seem necessary to do it separately.